docs/docs/Best_Practices/Runtime_Environment.md

```diff
@@ -0,0 +1,62 @@
+# Plugin Runtime Environment
+
+## Process Lifetime
+Plugin code runs inside of a Python interpreter process on the Delphix Engine.
+
+A fair question to ask is "What is the lifetime of this interpreter process?"  After all, if the interpreter
+process runs for a long time, then the plugin might be able to store things in memory for later access.
+
+Unfortunately, **there are no guarantees about process lifetime**. Your interpreter process could last two years, or it could last 400 microseconds. There is no way to know or predict this ahead of time.
+
+So, do not make any assumptions about interpreter process lifetime in your plugin code.
+
+
+## Available Modules
+Our Python 2.7 runtime environment only contains the [Python Standard Library](https://docs.python.org/2/library/). No additional Python modules/libraries are available.
+
+If you want to use some Python module that is not part of the standard library, you might be able to do so.
+You would need to include that library as part of your plugin. That would involve downloading the source
+code for that module, and copying it into your source directory.
+
+### Warnings
+There are two major things to watch out for if you decide to incorporate a 3rd-party library.
+
+1) Make sure you're legally allowed to do so! The licensing agreement on the module will decide if, and
+under what circumstances, you're allowed to make copies of, and redistribute the module. Some modules will
+allow this, some will disallow this, and some will allow this for a fee.
+
+2) Some Python libraries include native code (often written in C or C++). There is no support for using
+such libraries with plugin code.  The reason for this is that native code needs to be
+specially compiled and built for the machine that it the library will be running on. And, unfortunately,
+the machine your plugin runs on (the Delphix Engine) is likely very different from the machine you use
+to develop and build your plugin.
```

Comment:
should we tell them that they can use any linux x86 compatible libraries?

Reply:
I was a little torn on this.  I thought there were probably two good options:

1) Leave it vague like this. This gives enough clues to people who know what they're doing so that they know that they need to match architectures of binaries. This doesn't really help people who don't already know about binary packages and/or cross-compilation of source code, though.

2) Put in a ton more information, including examples. This would hopefully give everyone enough information to do what they need to do.

For now, I thought (1) was okay. I figured that we were likely going to be closing in on a real supported solution for 3rd-party libraries soon... and at that point we could expand this doc into something more generally useful.

What do you think?